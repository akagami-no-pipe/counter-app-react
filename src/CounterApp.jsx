import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value);

    const sumaUno = () => {
        setCounter(counter + 1);
        // setCounter((c) => c + 1);
    };

    const reset = () => {
        setCounter(value);
        // setCounter((c) => c - 1);
    };

    const restaUno = () => {
        setCounter(counter - 1);
        // setCounter(value);
    };
    
    return <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button onClick={sumaUno}>+1</button>
        <button aria-label='btn-reset' onClick={reset}>reset</button>
        <button onClick={restaUno}>-1</button>
    </>
};

CounterApp.propTypes = {
    value: PropTypes.number,
};

