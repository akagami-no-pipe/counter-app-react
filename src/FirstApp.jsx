import PropTypes from 'prop-types';


export const FirstApp = ({title = 'Sin titulo', subtitle = 'Sin subtitulo', name = 'Este es un nombre'}) => {
  // if(!title) throw new Error('El titulo es obligatorio');
  return (
    <>
      <h1 data-testid='test-title'> {title} </h1>
      <h2>{subtitle}</h2>
      <h2>{subtitle}</h2>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}