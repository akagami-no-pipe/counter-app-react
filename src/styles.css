
html, body{
    background-color: rgb(0, 47, 85);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}