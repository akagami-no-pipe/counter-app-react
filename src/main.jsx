

import React from 'react';
import ReactDOM from 'react-dom/client';
// apps
import HelloWorldApp from './HelloWorldApp'
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
// styles
import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={100}/>
    </React.StrictMode>
)