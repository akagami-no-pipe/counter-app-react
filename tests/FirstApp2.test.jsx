import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en componente firstapp', () => {
    
    const title = 'Hola mi gente!';
    const subtitle = 'Soy le mejor subtitulo';

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "hola mi gente"', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
    });

    test('debe mostrar el titul en un h1', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle}/>);
        expect(screen.getAllByText(subtitle).length).toBe(2);
    });

});