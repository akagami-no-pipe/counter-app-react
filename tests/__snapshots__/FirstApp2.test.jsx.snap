// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente firstapp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola mi gente!
     
  </h1>
  <h2>
    Sin subtitulo
  </h2>
  <h2>
    Sin subtitulo
  </h2>
  <p>
    Este es un nombre
  </p>
</div>
`;
